app/src/main/java/com/example/bookwormadventuresdeluxe2/MyProfileFragment.java

```diff
@@ -163,9 +186,54 @@ public void onClick(View view)
         }
     }
 
+    public void changeAuthEmail(EditText inputEmail) {
+        FirebaseUser user = FirebaseAuth.getInstance().getCurrentUser();
+
+        user.updateEmail(inputEmail.getText().toString())
+                .addOnCompleteListener(new OnCompleteListener<Void>() {
+                    @Override
+                    public void onComplete(@NonNull Task<Void> task) {
+                        if (task.isSuccessful()) {
+                            Log.d(TAG, "User email address updated.");
+                        }
+                        else
+                        {
+                            try
+                            {
+                                String errorCode = "";
+                                errorCode = ((FirebaseAuthException) task.getException()).getErrorCode();
+
+                                switch (errorCode)
+                                {
+                                    case "ERROR_INVALID_EMAIL":
+                                        /* Set Email EditText error code and additionally check password eligibility */
+                                        EditTextValidator.invalidEmail(inputEmail);
+                                        break;
+
+                                    case "ERROR_EMAIL_ALREADY_IN_USE":
+                                        EditTextValidator.emailTaken(inputEmail);
+                                        break;
+
+                                    default:
+                                        /* Unexpected Error code*/
+                                        throw new Exception("Unexpected Firebase Error Code"
+                                                + "inside click listener.");
+                                }
+                            }
+                            catch (Exception e)
+                            {
+                                /* Log message to debug*/
+                                Log.d(TAG, e.getMessage());
```

Comment:
Is this only triggered by the thrown exception in the default case? If so I would remove the try/catch because firebase error codes might not necessarily be a programming error (for example, what if we lose connection to firebase?)

Instead I would do:

default:
    Log.d(Tag, "Unexpected Firebase Error code: " + errorCode);

This code might not be syntactically correct, but you get the idea.